reddit_liveupdate/tests/eventcollector_tests.py

```diff
@@ -0,0 +1,153 @@
+import datetime
+import pytz
+from mock import MagicMock
+
+from pylons import app_globals as g
+
+from r2.tests import RedditTestCase, MockEventQueue
+from r2.lib import eventcollector
+from reddit_liveupdate.lib import events
+
+FAKE_DATE = datetime.datetime(2005, 6, 23, 3, 14, 0, tzinfo=pytz.UTC)
+
+
+class TestEventCollector(RedditTestCase):
+    def setUp(self):
+        self.liveupdate_events = events.LiveUpdateEventQueue()
+        self.domain_mock = self.autopatch(eventcollector, "domain")
+        self.autopatch(
+            self.liveupdate_events, "queue_production", MockEventQueue()
+        )
+        self.autopatch(
+            self.liveupdate_events, "queue_test", MockEventQueue()
+        )
+
+        self.created_ts_mock = MagicMock(name="created_ts")
+        self._datetime_to_millis = self.autopatch(
+            events, "_datetime_to_millis",
+            return_value=self.created_ts_mock)
+
+        self.context = MagicMock(name="context")
+        self.request = MagicMock(name="request")
+        self.liveevent = MagicMock(name="liveevent", _date=FAKE_DATE)
+
+        g.live_config["troublemaker_subreddits"] = []
+
+    def make_payload(self, **additional):
+        payload = {
+            "liveevent_id": self.liveevent._id,
+            "liveevent_fullname": self.liveevent._fullname,
+            "liveevent_title": self.liveevent.title,
+            "liveevent_description": self.liveevent.description,
+            "liveevent_created_ts": self.created_ts_mock,
+            "liveevent_banned": self.liveevent.banned,
+            "liveevent_banned_by": self.liveevent.banned_by,
+            "liveevent_nsfw": self.liveevent.nsfw,
+
+            'user_id': self.context.user._id,
+            'user_name': self.context.user.name,
+
+            'geoip_country': self.context.location,
+            'oauth2_client_id': self.context.oauth2_client._id,
+            'oauth2_client_app_type': self.context.oauth2_client.app_type,
+            'oauth2_client_name': self.context.oauth2_client.name,
+            'referrer_domain': self.domain_mock(),
+            'referrer_url': self.request.headers.get(),
+            'domain': self.request.host,
+            'user_agent': self.request.user_agent,
+            'user_agent_parsed': {
+                'platform_version': None,
+                'platform_name': None,
+            },
+            'obfuscated_data': {
+                'client_ip': self.request.ip,
+            },
+        }
+        payload.update(additional)
+        return payload
+
+    def test_create(self):
+        g.live_config["events_collector_liveupdate_create_sample_rate"] = 1.0
+        self.liveupdate_events.create_event(
+            self.liveevent,
+            context=self.context,
+            request=self.request
+        )
+        self.liveupdate_events.queue_production.assert_event_item(
+            dict(
+                event_topic="liveupdate_events",
+                event_type="ss.create_event",
```

Comment:
I would say the `event_type`s might be better as more explicit (say, "live_create" and "live_update") because as much as the aren't designed for it, people with still filter on only `event_type`